apps/test/unit/StudioAppTest.js

```diff
@@ -0,0 +1,71 @@
+import sinon from 'sinon';
+import {expect} from '../util/configuredChai';
+import {singleton as studioApp, stubStudioApp, restoreStudioApp} from '@cdo/apps/StudioApp';
+import {throwOnConsoleErrors, throwOnConsoleWarnings} from '../util/testUtils';
+import {assets as assetsApi} from '@cdo/apps/clientApi';
+import {listStore} from '@cdo/apps/code-studio/assets';
+
+describe('StudioApp.singleton', () => {
+  throwOnConsoleErrors();
+  throwOnConsoleWarnings();
+
+  beforeEach(stubStudioApp);
+  afterEach(restoreStudioApp);
+
+  let containerDiv, codeWorkspaceDiv;
+  beforeEach(() => {
+    codeWorkspaceDiv = document.createElement('div');
+    codeWorkspaceDiv.id = 'codeWorkspace';
```

Comment:
yes. That fix was to unbreak other tests. The issue is that even after this test finishes (and removes the codeWorkspace element it created), the event handler remains attached to the window and fires in *other* tests.

What we really need is a proper teardown method for StudioApp that unregisters the various global listeners it created.